src/event_listener.rs

```diff
@@ -33,7 +49,7 @@ pub trait EntityEvent: Event + Clone {
 pub struct On<E: EntityEvent> {
     phantom: PhantomData<E>,
     /// A function that is called when the event listener is triggered.
-    pub(crate) callback: CallbackSystem,
+    pub(crate) callbacks: Vec<(ListenerHandle<E>, CallbackSystem)>,
```

Comment:
This is probably clearer as a wrapped type, rather than a tuple struct. Create a struct that stores handle + system.

Reply:
This should actually be kept as a tuple; we're using a map-like pattern, but can't use an ordinary map data structure because we need stable order. Comment here instead.